### PR TITLE
Re-balances ant damage values & lets you pour ants on the floor

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
One final change I wanted to make to ants was to rework their rather low damage values. While they by no means were ever intended to be a kill-chem, the fact that they only did 25 brute after being injected with 200u was just far too low.
To change this, I've reworked their damage formula, so now you do roughly 32 brute damage with 100u of ants. This scales significantly once you add more ants after that point.
This is still a rather low amount of damage for the amount of u it takes, so to balance that I made the chance for one to vomit out the ants also stun the user, however to balance that balance it will purge anywhere from 5-10u of ants upon doing so.
I have also changed how the debuff works somewhat. While it still does its normal damage over time, I have increased the damage done by the user scratching at themselves, which yielded an average of 100 brute when dumped with 200u ants during testing.
Finally, you can pour ants back onto the ground now, with the amount of ants being poured being directly proportional to the amount of damage they will do. You must pour more than 10 ants for it to appear, and the amount of ants will only change the maximum damage that the resulting caltrop will do, not the minimum damage. There is a hard-cap at a 10 damage maximum, with the minimum always being 0.5. The amount of damage this caltrop does is equal to `ant_volume * 0.2`
Finally to balance this a little more, breeding ants has gone from requiring 6u sugar to 8u sugar, making farming take a little longer.

I hope that this makes it more fun to see what you can do with ants.

## Why It's Good For The Game
Ants are far too weak in their current state, even for a chem that's not meant to kill in smaller unit numbers. I hoped to alleviate this while not making them too overpowered or underpowered.

## Changelog
:cl: Wallem
add: You can now pour ants back onto the ground, with the more ants you pour leading to more damage! 
balance: Ants now do more damage, however they require more sugar in order to breed.
imageadd: The ant decal now has a couple shiny new sprites!
/:cl:

### CODE REVIEW

**Comment by AndroidMK2** [comment]
If I like inject pure sugar to a rotting mess on the maint floor it will accelerate their breeding?

**Comment by Wallemations** [comment]
> If I like inject pure sugar to a rotting mess on the maint floor it will accelerate their breeding?

Rotting messes don't do anything, the only way ants breed are by putting sugar in a beaker with them.
While I say "breeding" it's really just a chem recipe.

**Comment by moocowswag** [comment]
Will large amounts of ant smoke and ant foam leave ants on the floor aswell?

**Comment by maxymax13** [comment]
can you make it so there's a way to get past piles of shit made by maint spawned food without getting 5 million brute damage on my legs because i You Stepped On The Plating! (x4), if you make them spillable then now you can't even get rid of them with a beaker, give botanists some sort of ant murder device

**Comment by Wallemations** [comment]
> can you make it so there's a way to get past piles of shit made by maint spawned food without getting 5 million brute damage on my legs because i You Stepped On The Plating! (x4), if you make them spillable then now you can't even get rid of them with a beaker, give botanists some sort of ant murder device

Maint-spawned food shouldn't spawn ants. There is a system in place that makes maploaded foods only able to decompose when they are touched.
Yes you can still get rid of them with a beaker, or with space cleaner, or a mop, etc.

**Comment by TemporalOroboros** [comment]
> > can you make it so there's a way to get past piles of shit made by maint spawned food without getting 5 million brute damage on my legs because i You Stepped On The Plating! (x4), if you make them spillable then now you can't even get rid of them with a beaker, give botanists some sort of ant murder device
> 
> Maint-spawned food don't spawn ants. There is a system in place that makes maploaded foods only able to decompose when they are touched.
> Yes you can still get rid of them with a beaker, or with space cleaner, or a mop, etc. They're still cleanable decals as well as scoopable.

I'm pretty sure the maint ants they're running into are a result of mice getting electrocuted and rotting.

**Comment by RaveRadbury** [comment]
> I'm pretty sure the maint ants they're running into are a result of mice getting electrocuted and rotting.

How will the lizards survive with their only reliable food source rotting away in maint? 😢

**Comment by Wallemations** [comment]
> Will large amounts of ant smoke and ant foam leave ants on the floor aswell?

There is a 10u threshhold on creating ants on the floor, anything less than 10u will do nothing. From my testing neither smoke nor foam could create ants this way.

**Comment by Wallemations** [comment]
> > > can you make it so there's a way to get past piles of shit made by maint spawned food without getting 5 million brute damage on my legs because i You Stepped On The Plating! (x4), if you make them spillable then now you can't even get rid of them with a beaker, give botanists some sort of ant murder device
> > 
> > 
> > Maint-spawned food don't spawn ants. There is a system in place that makes maploaded foods only able to decompose when they are touched.
> > Yes you can still get rid of them with a beaker, or with space cleaner, or a mop, etc. They're still cleanable decals as well as scoopable.
> 
> I'm pretty sure the maint ants they're running into are a result of mice getting electrocuted and rotting.

I genuinely hadn't thought of that. I'll see if I can make some kind of workaround.

**Comment by Wallemations** [comment]
There, that should fix the mouse problem too.

**Comment by KittyNoodle** [comment]
>  From my testing neither smoke nor foam could create ants this way.

Chemical payload with bs beakers.

**Comment by BebeYoshi** [comment]
The fact that I was stung by multiple ant bees at the same time and multiple times by each one, I ended up with 300 units and didn't even go near crit, this is a good change.

**Comment by Wallemations** [comment]
With ant caltrops, roughly 40u of ants makes the `max_damage` roughly 8. If people feel this is too much, I can make the decimal value that calculates ant bite damage smaller.

**Comment by Wallemations** [comment]
> > From my testing neither smoke nor foam could create ants this way.
> 
> Chemical payload with bs beakers.

Genuinely forgot those exist. With admin commands I can make grenades out of anything so I'll see what 299u of ants with 1u foam does and report back.

**Comment by Wallemations** [comment]
> > > From my testing neither smoke nor foam could create ants this way.
> > 
> > 
> > Chemical payload with bs beakers.
> 
> Genuinely forgot those exist. With admin commands I can make grenades out of anything so I'll see what 299u of ants with 1u foam does and report back.

You need about 280u ants with 20u foam to get any ants at all. Admittedly they have a pretty high damage number for them, so I think I'll lower that.

**Comment by Wallemations** [comment]
Gonna try and figure out a way to visually indicate how many ants are in a pile so you can guess how much damage they'll do to you.

**Comment by Wallemations** [comment]
Converting to draft while I either resprite the ant decal to visually show how many ants are there or get someone to do it for me.

**Comment by Wallemations** [comment]
NEVERMIND JUST FOUND OUT THERE WAS A FEATURE FREEZE
THIS PR WAS OPEN BEFORE DON'T DELETE IT PLS THANKS

**Comment by Wallemations** [comment]
Oh wait it doesn't show up above the feature freeze thing.
Neat

**Comment by TheBonded** [comment]
> The fact that I was stung by multiple ant bees at the same time and multiple times by each one, I ended up with 300 units and didn't even go near crit, this is a good change.

Ant bees.

**Comment by Wallemations** [comment]
https://user-images.githubusercontent.com/66052067/129459021-b18a5a26-58fc-473f-8751-fd8d70789f9b.mp4

Here's the new ant sprites, hopefully they'll both be easier to scoop up as well as be useful in showing how much damage a pile of ants will do.

**Comment by Wallemations** [comment]
I was told that the anthill should grow the more ants there are, so I attempted to do so.

**Comment by Wallemations** [comment]
Ok so RemoveComponent doesn't exist, meaning I need to change the value of the component manually. Time to fix this I guess.

**Comment by Wallemations** [comment]
Alright I fixed it up, now there's a new flag & signal for updating caltrop damage values after the caltrop has already been made. If a caltrop has the flag, then it registers the signal, and you just need to send that signal with the new `min_damage` and `max_damage` values.
Idk what other caltrops would use that besides ants but it's useful in the future I guess lmao

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -1414,3 +1414,7 @@
 
 /// Called on the organ when it is removed from someone (mob/living/carbon/old_owner)
 #define COMSIG_ORGAN_REMOVED "comsig_organ_removed"
+
+// Caltrop Signals
+/// Called when a caltrop with the CALTROP_UPDATE_DAMAGE flag gets its damage updated. For example, (/obj/effect/decal/cleanable/ants)
+#define COMSIG_CALTROP_UPDATE "comsig_caltrop_update"
```
Comment (AMonkeyThatCodes) [inline code/__DEFINES/dcs/signals.dm:1420]: this signal is shit. it's a proc disguised as a signal.

File: code/datums/components/food/decomposition.dm [hunk 2]
```diff
@@ -23,12 +23,12 @@
 	/// Used for examining
 	var/examine_type = DECOMP_EXAM_NORMAL
 
-/datum/component/decomposition/Initialize(mapload, decomp_flags = NONE)
+/datum/component/decomposition/Initialize(mapload, semi_preserved_food, decomp_flags = NONE)
```
Comment (AMonkeyThatCodes) [inline code/datums/components/food/decomposition.dm:26]: these vars are awful. they dont describe anything. What the hell is a semi_preserved food?

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (Wallemations) [reply code/__DEFINES/dcs/signals.dm:1420]: How else do I send a proc to a component?

File: code/datums/components/food/decomposition.dm (hunk 2 above)
Comment (Wallemations) [reply code/datums/components/food/decomposition.dm:26]: It's explained in `code/game/objects/items/food/_food.dm`:
```
///Food that needs to be picked up in order to decompose.
var/semi_preserved_food = FALSE
```

Comment (AMonkeyThatCodes) [reply code/datums/components/food/decomposition.dm:26]: my point being that the name should be self-explanatory.

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (AMonkeyThatCodes) [reply code/__DEFINES/dcs/signals.dm:1420]: you shouldn't.  https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674

File: code/datums/components/food/decomposition.dm (hunk 2 above)
Comment (Wallemations) [reply code/datums/components/food/decomposition.dm:26]: I couldn't come up with a better name for the var, what would you recommend for something like that?

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (Wallemations) [reply code/__DEFINES/dcs/signals.dm:1420]: Reading through the [new location where the guide is](https://hackmd.io/@tgstation/SignalsComponentsElements) according to the original author of it, it really just points towards signals being the way to interact with a component or element, as far as I can tell.
I can't really think of any other way to transfer a change in a variable in one object which can be manipulated by a player into a component other than a signal, but I am always one to admit my rather amateur coding knowledge, so if there is another way to do this then I'm not privy to it yet.

**Review by Twaticus — APPROVED** [review]

File: code/datums/components/food/decomposition.dm (hunk 2 above)
Comment (Wallemations) [reply code/datums/components/food/decomposition.dm]: There, I think that variable better explains it.

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (AMonkeyThatCodes) [reply code/__DEFINES/dcs/signals.dm:1431]: we can discuss this more in-depth in discord if you'd like. that might be easier

Comment (Watermelon914) [reply code/__DEFINES/dcs/signals.dm:1431]: bad usage of signals. Use the COMPONENT_DUPE_UNIQUE_PASSARGS on components and then just simply add the component again with the new damage value.
Then you can override /datum/component/InheritComponent to update your component.

**Comment by LemonInTheDark** [comment]
@Wallemations bug me/some maint when this is ready yeah?

**Comment by Wallemations** [comment]
> @Wallemations bug me/some maint when this is ready yeah?

Sure thing, I apologize for the long delay, college started up recently and I've been swamped trying to piece a lot of stuff together.

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (Wallemations) [reply code/__DEFINES/dcs/signals.dm:1431]: Oh yea looking at it, this is exactly what I needed. I completely forgot that components had `dupe_mode` as a variable, thanks for reminding me of this.

**Comment by Wallemations** [comment]
Well I know you just drafted this but I genuinely can't think of anything else that might be a problem with ants at the moment. So I guess it's ready!

**Pull request merged**